pkg/util/clusterdata/worker_profiles_feature.go

```diff
@@ -0,0 +1,107 @@
+package clusterdata
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	azureproviderv1beta1 "github.com/openshift/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+	clusterapi "github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+const (
+	workerMachineSetsNamespace = "openshift-machine-api"
+)
+
+func newWorkerProfilesEnricherFeature(restConfig *rest.Config, oc *api.OpenShiftCluster) (enricherFeature, error) {
+	client, err := clusterapi.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &workerProfilesEnricherFeature{
+		client: client,
+		oc:     oc,
+	}, nil
+}
+
+type workerProfilesEnricherFeature struct {
+	client clusterapi.Interface
+	oc     *api.OpenShiftCluster
+}
+
+func (ef *workerProfilesEnricherFeature) Name() string {
+	return "worker profiles"
+}
+
+func (ef *workerProfilesEnricherFeature) FetchData(ctx context.Context) (func(), error) {
+	r, err := azure.ParseResourceID(ef.oc.ID)
+	if err != nil {
+		return ef.setDefaults, err
+	}
+
+	machinesets, err := ef.client.MachineV1beta1().MachineSets(workerMachineSetsNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return ef.setDefaults, err
+	}
+
+	if len(machinesets.Items) == 0 {
+		return ef.setDefaults, fmt.Errorf("worker machinesets not found in %q", workerMachineSetsNamespace)
+	}
+
+	workerProfiles := make([]api.WorkerProfile, len(machinesets.Items), len(machinesets.Items))
+	for i, machineset := range machinesets.Items {
+		var machineProviderSpec azureproviderv1beta1.AzureMachineProviderSpec
+		_, _, err := codecs.UniversalDeserializer().Decode(machineset.Spec.Template.Spec.ProviderSpec.Value.Raw, nil, &machineProviderSpec)
```

Comment:
Check if `machineset.Spec.Template.Spec.ProviderSpec.Value.Object` can be used instead.

Reply:
We can't use it, because it is `nil`.